followlinks.go

```diff
@@ -122,6 +134,80 @@ func (r *symlinkResolver) readSymlink(p string, allowWildcard bool) ([]string, e
 	}, nil
 }
 
+func readFile(fs FS, root string) (os.DirEntry, error) {
+	var out os.DirEntry
+
+	root = filepath.Clean(root)
+	if root == "/" || root == "." {
+		return nil, nil
+	}
+
+	err := fs.Walk(context.TODO(), root, func(p string, entry os.DirEntry, err error) error {
```

Comment:
is `fs.Walk` called on non-directory here?

Reply:
As discussed, this should be fine, since we're just doing the same behavior from the go standard library.